src/qt/optionsmodel.cpp

```diff
@@ -637,4 +617,49 @@ void OptionsModel::checkAndMigrate()
     if (settings.contains("addrSeparateProxyTor") && settings.value("addrSeparateProxyTor").toString().endsWith("%2")) {
         settings.setValue("addrSeparateProxyTor", GetDefaultProxyAddress());
     }
+
+    // Migrate and delete legacy GUI settings that have now moved to <datadir>/settings.json.
```

Comment:
This migration may silently drop configs from `.config/Bitcoin/Bitcoin-Qt.conf` with unexpected effects. Before this PR, values in `Bitcoin-Qt.conf` were overriding values from `bitcoin.conf`. So, for example, if one had:

`bitcoin.conf`: `listen=1`
`Bitcoin-Qt.conf `: `fListen=false`

then `bitcoin-qt` would not be listening and "Allow incoming connections" will be unchecked in the GUI (correct).

However, once the migration happens, it would leave the configs in this state:

`bitcoin.conf`: `listen=1`
`Bitcoin-Qt.conf `: (no fListen is present)

which will cause `bitcoin-qt` to start listening, silently, when it was not listening before.

Reply:
Thanks for the elaborate explanation! I was wrong, sorry for the noise.

Can be marked as resolved.